cmd/dotmesh-server/pushinitiatorstate.go

```diff
@@ -496,10 +497,46 @@ func (f *fsMachine) retryPush(
 					return &Event{
 						Name: "peer-up-to-date",
 					}, backoffState
+				case *ToSnapsDiverged:
+					if transferRequest.StashDivergence {
+						returnErr = false
+						var newBranch string
+						e := client.CallRemote(
+							ctx,
+							"DotmeshRPC.StashAfter",
+							types.StashRequest{
+								FilesystemId: toFilesystemId,
+								SnapshotId:   err.latestCommonSnapshot.Id,
+							},
+							&newBranch,
+						)
+						if e != nil {
+							return &Event{
+								Name: "failed-stashing-remote-end", Args: &EventArgs{"err": e},
+							}, backoffState
+						}
+						// could also re-fetch the remote end commits here
+						remoteSnaps, e := restrictSnapshots(remoteSnaps, err.latestCommonSnapshot.Id)
```

Comment:
I'm wondering if here onward I should just crash out and return an error, which would cut this duplicate code from above and mean we just go round again. Thoughts @alaric-dotmesh ?

Reply:
otherwise I'm going to either need to copy and paste it to the `toSnapsAhead` case I'm working on or make another function for it